tests/ttnn/distributed/test_distributed_reshape.cpp

```diff
@@ -0,0 +1,281 @@
+    }
+    if (num_devices < 8 or arch != tt::ARCH::WORMHOLE_B0) {
+        GTEST_SKIP() << "Skipping T3K Multi-Device test suite on non T3K machine.";
+    }
+}
+
+std::vector<chip_id_t> get_physical_device_ids(const MeshDevice& mesh) {
+    std::vector<chip_id_t> device_ids;
+    for (auto* device : mesh.get_devices(ttnn::distributed::MeshType::RowMajor)) {
+        device_ids.push_back(device->id());
+    }
+    return device_ids;
+}
+
+static constexpr std::array<MeshShape, 24> kMeshShapes{
+    {{1, 1}, {1, 2}, {1, 3}, {1, 4}, {1, 5}, {1, 6}, {1, 7}, {1, 8}, {2, 1}, {2, 2}, {2, 3}, {2, 4},
+     {3, 1}, {3, 2}, {4, 1}, {4, 2}, {8, 1}, {7, 1}, {6, 1}, {5, 1}, {4, 1}, {3, 1}, {2, 1}, {1, 1}}};
+
+class MeshConfigurationTest : public ::testing::TestWithParam<MeshShape> {
+protected:
+    void SetUp() override { check_test_environment(); }
+};
+
+TEST_P(MeshConfigurationTest, TestMeshConfigurations) {
+    const auto& shape = GetParam();
+    auto mesh = ttnn::distributed::open_mesh_device(
+        {shape.num_rows, shape.num_cols},
+        DEFAULT_L1_SMALL_SIZE,
+        DEFAULT_TRACE_REGION_SIZE,
+        1,
+        tt::tt_metal::DispatchCoreType::WORKER);
+    EXPECT_EQ(mesh->num_rows(), shape.num_rows);
+    EXPECT_EQ(mesh->num_cols(), shape.num_cols);
+    ttnn::distributed::close_mesh_device(mesh);
+}
+
+// Test all possible mesh configurations on T3000
+INSTANTIATE_TEST_SUITE_P(MeshShapes, MeshConfigurationTest, ::testing::ValuesIn(kMeshShapes));
+
+class MeshReshapeTest : public ::testing::TestWithParam<std::tuple<MeshShape, MeshShape>> {
+protected:
+    void SetUp() override { check_test_environment(); }
+};
+
+TEST_P(MeshReshapeTest, TestReshapeBetweenConfigurations) {
+    const auto& [old_shape, new_shape] = GetParam();
+
+    if ((old_shape.num_rows * old_shape.num_cols) != (new_shape.num_rows * new_shape.num_cols)) {
+        GTEST_SKIP() << "Device counts don't match; we test this in InvalidReshapeDimensions";
+    }
+    if (old_shape.num_rows == 1 or old_shape.num_cols == 1) {
+        GTEST_SKIP() << "Old shape is 1xN or Nx1; we test this in From1x4To2x2Invalid";
+    }
+
+    auto mesh = ttnn::distributed::open_mesh_device(
+        {old_shape.num_rows, old_shape.num_cols},
+        DEFAULT_L1_SMALL_SIZE,
+        DEFAULT_TRACE_REGION_SIZE,
+        1,
+        tt::tt_metal::DispatchCoreType::WORKER);
+
+    EXPECT_EQ(mesh->num_rows(), old_shape.num_rows);
+    EXPECT_EQ(mesh->num_cols(), old_shape.num_cols);
+
+    auto original_order = get_physical_device_ids(*mesh);
+
+    // Attempt reshape
+    mesh->reshape({new_shape.num_rows, new_shape.num_cols});
+
+    // Verify new shape
+    EXPECT_EQ(mesh->num_rows(), new_shape.num_rows);
+    EXPECT_EQ(mesh->num_cols(), new_shape.num_cols);
+
+    // Verify device ordering is preserved
+    auto new_order = get_physical_device_ids(*mesh);
+    EXPECT_EQ(original_order, new_order);
```

Comment:
couple of nits: can get rid of the extra variables, and directly compare to `original_order`. Also the order of comparison logically should be flipped ("expect new order equal to original order", not the other way):
```
EXPECT_EQ(get_physical_device_ids(*mesh), original_order);